### PR TITLE
Add sleep to ensure that etcd is added to reconcile queue

### PR DESCRIPTION
**Pull request by georgekuruvillak**
**What this PR does / why we need it**:
This PR adds a wait to ensure that etcd is reconciled and statefulset is deleted as part of reconcilation.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: controllers/etcd_controller_test.go [hunk 1]
```diff
@@ -229,6 +229,8 @@ var _ = Describe("Druid", func() {
 			setStatefulSetReady(&sts)
 			err = c.Status().Update(context.TODO(), &sts)
 			Expect(err).NotTo(HaveOccurred())
+			// sleep needed here as we need to wait till etcd resource is added to the queue
+			time.Sleep(time.Second * 2)
```
Comment (swapnilgm) [inline controllers/etcd_controller_test.go:233]: looks like you will have to increase the time here.

Comment (georgekuruvillak) [reply controllers/etcd_controller_test.go]: Changed the logic to get rid of the sleep. It was dirty anyway :D

File: controllers/etcd_controller_test.go [hunk 2]
```diff
@@ -279,6 +290,44 @@ func etcdRemoved(c client.Client, etcd *druidv1alpha1.Etcd) error {
...
+	if err := c.Get(ctx, req, e); err != nil {
+		return err
+	}
+	if e.Status.Ready == nil || !*e.Status.Ready {
+		return fmt.Errorf("etcd not ready")
+	}
+	return nil
+}
+
+func setandCheckStatefulSetReady(c client.Client, etcd *druidv1alpha1.Etcd) error {
```
Comment (amshuman-kr) [inline controllers/etcd_controller_test.go:310]:
```suggestion
func setAndCheckStatefulSetReady(c client.Client, etcd *druidv1alpha1.Etcd) error {
```

File: controllers/etcd_controller_test.go [hunk 3]
```diff
@@ -226,9 +227,19 @@ var _ = Describe("Druid", func() {
...
 			Eventually(func() error { return statefulsetIsCorrectlyReconciled(c, instance, &sts) }, timeout, pollingInterval).Should(BeNil())
-			setStatefulSetReady(&sts)
-			err = c.Status().Update(context.TODO(), &sts)
-			Expect(err).NotTo(HaveOccurred())
+			// This go-routine is to set that statefulset is ready manually as statefulset controller is absent for tests.
+			go func() {
+				for {
+					select {
+					case <-time.After(time.Second * 2):
+						setandCheckStatefulSetReady(c, instance)
```
Comment (amshuman-kr) [inline controllers/etcd_controller_test.go:235]:
```suggestion
						setAndCheckStatefulSetReady(c, instance)
```

**Comment by georgekuruvillak** [comment]
> Overall looks fine. Just one naming suggestion.
> 
> One question though. Doesn't the [`time.After`](https://github.com/gardener/etcd-druid/pull/49/files#diff-03838f4ddc16876e7feda81a4bedf7d8R234) suffer from the same problem as sleep? What if it takes more than `2s`? Or am I missing something?

As discussed offline, the `time.After` is in a for loop.

**Pull request merged**